dev/src/reference.ts

```diff
@@ -1940,6 +1992,19 @@ export class Query {
       return 0;
     };
   }
+
+  /**
+   * Applies a custom data converter to this Query, allowing you to use your
+   * own custom model objects with Firestore. When you call get() on the
+   * returned Query, the provided converter will convert between Firestore
+   * data and your custom type U.
+   *
+   * @param converter Converts objects to and from Firestore.
+   * @return A Query<U> that uses the provided converter.
+   */
+  withConverter<U>(converter: FirestoreDataConverter<U>): Query<U> {
+    return new Query<U>(this.firestore, this._queryOptions, converter);
```

Comment:
Here and everywhere: Should we validate that a `converter` is provided?

Reply:
Would this be for checking JS code? We don't currently validate converters in the web SDK. Do we usually verify all user inputs beyond the type system?